### PR TITLE
Finished prototype for the Profile pages. 

### PR DESCRIPTION
**Pull request by skq8614**
The main changes can be found in ProfileServlet.java and profile.jsp. Now, whenever a user is listed in a conversation you can click on their username and it will take them to that user's profile page. If you are not that user, the profile page will simply say "this is the profile page for ____." If you are the user looking at your own page, the profile page will say "this is your page." Each profile has a unique URL which is localhost:8080/users/_____, the blank being the username of whoever's profile you are viewing. I have yet to figure out how to correctly complete the Unit tests for my ProfileServlet, so for now that file is commented out.

### CODE REVIEW

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+        String requestUrl = request.getRequestURI();
+        String user = requestUrl.substring("/users/".length());
+        request.setAttribute("user", user);
+        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
+    }
+
+    /**
+     * Currently does nothing. Will update for MVP version of profile.
```
Comment (dgalbraith33) [inline src/main/java/codeu/controller/ProfileServlet.java]: What will this do in the MVP?

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 2]
```diff
@@ -58,6 +58,10 @@ public static UserStore getTestInstance(PersistentStorageAgent persistentStorage
   /** The in-memory list of Users. */
   private List<User> users;
 
+  public List<User> getAllUsers() {
+        return users;
```
Comment (dgalbraith33) [inline src/main/java/codeu/model/store/basic/UserStore.java]: nit: Can you fix this indentation? Should be 2 spaces for top level and 4 for second level.

File: src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -66,4 +66,15 @@
     <url-pattern>/register</url-pattern>
   </servlet-mapping>
 
+  <servlet>
+      <servlet-name>ProfileServlet</servlet-name>
```
Comment (dgalbraith33) [inline src/main/webapp/WEB-INF/web.xml]: Remove one indentation level for this and the ones below.

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Test
+    public void testDoGet() throws IOException, ServletException {
+        profileServlet.doGet(mockRequest, mockResponse);
+        Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+    }
+    */
+
+}
```
Comment (dgalbraith33) [inline src/test/java/codeu/controller/ProfileServletTest.java]: nit: please add a newline at the end of the file

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class ProfileServletTest {
+    /*
+    private ProfileServlet profileServlet;
```
Comment (dgalbraith33) [inline src/test/java/codeu/controller/ProfileServletTest.java]: Can you uncomment/fix these tests?

**Review by dgalbraith33 — CHANGES_REQUESTED** [review]

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 1 above)
Comment (tiffanycwong) [reply src/main/java/codeu/controller/ProfileServlet.java]: A good convention is to add a TODO tag
E.g. TODO: <<Insert what implement for MVP>>

Comment (tiffanycwong) [reply src/main/java/codeu/controller/ProfileServlet.java]: Oops I meant to start a review

File: src/main/java/codeu/model/store/basic/UserStore.java (hunk 2 above)
Comment (tiffanycwong) [reply src/main/java/codeu/model/store/basic/UserStore.java]: Good habit to review your PR and diffs before adding reviewers so that small changes like these which can be difficult to catch in a text editor are cleaned up

File: src/main/webapp/index.jsp [hunk 6]
```diff
@@ -37,7 +37,7 @@
       style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
 
       <h1>CodeU Chat App</h1>
-      <h2>Welcome!</h2>
+      <h2>Welcome to our chat!</h2>
```
Comment (tiffanycwong) [inline src/main/webapp/index.jsp]: This change seems irrelevant to the rest of the PR. To build good habits, remove if unintended change. Otherwise, change in new PR.

Not a big deal here but generally speaking, we want to keep PRs scoped. For example, if profile page changes introduce an error, we could want to be able to revert this PR without reverting other unrelated changes.

File: src/test/java/codeu/controller/ProfileServletTest.java (hunk 5 above)
Comment (tiffanycwong) [reply src/test/java/codeu/controller/ProfileServletTest.java]: A good start for tests would be to check the message posted (if you have other ideas, that works too!)

**Review by tiffanycwong — CHANGES_REQUESTED** [review]
